src/linalg/nonbacktracking.jl

```diff
@@ -26,17 +26,20 @@ function non_backtracking_matrix(g::AbstractGraph)
         end
     end
 
-    B = zeros(Float64, m, m)
-
+    rowidx = Int[]
```

Comment:
Since we know the size we can allocate the full vector ahead of time and avoid pushing

Reply:
Thanks for your reviewing. I modified to allocate a capacity by `sizehint!`. The size `nz` is calculated as follows:

$$
\text{nz} \coloneqq \begin{cases}
\displaystyle
\sum_{u\in V} d_{u} (d_{u} - 1), & \text{graph is undirected}
\\
\displaystyle
\sum_{u\in V} d_{u}^{\text{in}} d_{u}^{\text{out}}, & \text{graph is directed}
\end{cases}
$$